public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataTransformationsTab.tsx

```diff
@@ -56,42 +60,70 @@ export class PanelDataTransformationsTab
 
 export function PanelDataTransformationsTabRendered({ model }: SceneComponentProps<PanelDataTransformationsTab>) {
   const styles = useStyles2(getStyles);
+  const sourceData = model.getQueryRunner().useState();
   const { data, transformations: transformsWrongType } = model.getDataTransformer().useState();
   // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
   const transformations: DataTransformerConfig[] = transformsWrongType as unknown as DataTransformerConfig[];
 
-  if (transformations.length < 1) {
-    return <EmptyTransformationsMessage onShowPicker={() => {}}></EmptyTransformationsMessage>;
-  }
+  const [drawerOpen, setDrawerOpen] = useState<boolean>(false);
+  const [confirmModalOpen, setConfirmModalOpen] = useState<boolean>(false);
 
-  if (!data) {
+  if (!data || !sourceData.data) {
     return;
   }
 
+  const transformationsDrawer = (
+    <TransformationsDrawer
+      onClose={() => setDrawerOpen(false)}
+      onTransformationAdd={(selected) => {
+        if (selected.value === undefined) {
+          return;
+        }
+        model.changeTransformations([...transformations, { id: selected.value, options: {} }]);
+        setDrawerOpen(false);
+      }}
+      isOpen={drawerOpen}
+      series={data.series}
+    ></TransformationsDrawer>
+  );
+
+  if (transformations.length < 1) {
+    return (
+      <>
+        <EmptyTransformationsMessage onShowPicker={() => setDrawerOpen(true)}></EmptyTransformationsMessage>
+        {transformationsDrawer}
+      </>
+    );
+  }
+
   return (
     <>
-      <TransformationsEditor data={data} transformations={transformations} model={model} />
+      <TransformationsEditor data={sourceData.data} transformations={transformations} model={model} />
```

Comment:
I realized this needs to be the untransformed data, otherwise series created by transformations that appear later in the chain will appear before they exist.